OpenDBDiff.Schema/Model/SchemaBase.cs

```diff
@@ -229,7 +229,7 @@ public virtual ObjectStatus Status
                     status = value;
 
                 // Si el estado de la tabla era el original, lo cambia, sino deja el actual estado.
-                if (Parent != null && Parent.Status == ObjectStatus.Original || value == ObjectStatus.Rebuild || value == ObjectStatus.RebuildDependencies)
+                if (Parent != null && (Parent.Status == ObjectStatus.Original || value == ObjectStatus.Rebuild || value == ObjectStatus.RebuildDependencies))

```

Comment:
I agree with this fix, but I'd like to rework the logic in this property a bit. But I still want you to get the credit. Would you mind changing the entire `SchemaBase.Status` property to:

```c#
        public virtual ObjectStatus Status
        {
            get { return status; }
            set
            {
                if (status != ObjectStatus.Rebuild && status != ObjectStatus.RebuildDependencies)
                    status = value;

                if (Parent == null) return;

                // Si el estado de la tabla era el original, lo cambia, sino deja el actual estado.
                // If the state of the table was the original, it changes it, but leaves the current state. (Google translated)
                if (Parent.Status == ObjectStatus.Original
                    || value == ObjectStatus.Rebuild
                    || value == ObjectStatus.RebuildDependencies)
                {
                    switch (value)
                    {
                        case ObjectStatus.RebuildDependencies:
                        case ObjectStatus.Rebuild:
                            Parent.Status = value;
                            break;

                        case ObjectStatus.Original:
                            break;

                        default:
                            Parent.Status = ObjectStatus.Alter;
                            break;
                    }
                }
            }
        }
```